plugins/org.eclipse.elk.alg.packing.rectangles/src/org/eclipse/elk/alg/packing/rectangles/firstiteration/Calculations.java

```diff
@@ -0,0 +1,213 @@
+
+import org.eclipse.elk.alg.packing.rectangles.util.DrawingData;
+import org.eclipse.elk.alg.packing.rectangles.util.DrawingDataDescriptor;
+import org.eclipse.elk.graph.ElkNode;
+
+/**
+ * Class that offers methods for small operations in the first iteration.
+ * 
+ * @author dalu
+ */
+public final class Calculations {
+
+    private Calculations() {
+    }
+
+    /**
+     * Calculates the drawing width for placement options LPR and LPB.
+     * 
+     * @param drawingWidth
+     *            width of drawing before placement of new rectangle.
+     * @param xCoord
+     *            potential x coordinate of new rectangle.
+     * @param toPlaceWidth
+     *            width of new rectangle.
+     * @return width of drawing including new rectangle in case LPR and LPB.
+     */
+    protected static double getWidthLPRorLPB(final double drawingWidth, final double xCoord,
+            final double toPlaceWidth) {
+        return Math.max(drawingWidth, xCoord + toPlaceWidth);
+    }
+
+    /**
+     * Calculates the drawing height for placement options LPR and LPB.
+     * 
+     * @param drawingHeight
+     *            height of drawing before placement of new rectangle.
+     * @param yCoord
+     *            potential y coordinate of new rectangle.
+     * @param toPlaceHeight
+     *            height of new rectangle.
+     * @return height of drawing including new rectangle in case LPR and LPB.
+     */
+    protected static double getHeightLPRorLPB(final double drawingHeight, final double yCoord,
+            final double toPlaceHeight) {
+        return Math.max(drawingHeight, yCoord + toPlaceHeight);
+    }
+
+    /**
+     * Calculates the y-coordinate after the shift of the rectangle to be placed right of lastPlaced.
+     * 
+     * @param xCoordRectToPlace
+     *            x-value of the rectangle to be placed.
+     * @param placedRects
+     *            a list of already placed rectangles.
+     * @param lastPlaced
+     *            rectangle that was placed most recently.
+     * @return y-coordinate after the shift of the rectangle to be placed right of lastPlaced
+     */
+    protected static double calculateYforLPR(final double xCoordRectToPlace, final List<ElkNode> placedRects,
+            final ElkNode lastPlaced) {
+        ElkNode closestUpperNeighbor = null;
+        double closestNeighborBottomBorder = 0;
+        // find neighbors that lay between the upper and lower border of the rectangle to be placed.
+        for (ElkNode placedRect : placedRects) {
+            double placedRectBottomBorder = placedRect.getY() + placedRect.getHeight();
+            if (verticalOrderConstraint(placedRect, xCoordRectToPlace)) {
+                // is closest neighbor?
+                if (closestUpperNeighbor == null) {
+                    closestUpperNeighbor = placedRect;
+                } else if (lastPlaced.getY() - placedRectBottomBorder < lastPlaced.getY()
+                        - closestNeighborBottomBorder) {
```

Comment:
Avoid such unfortunate line breaks. This is where auto formatting in Eclipse doesn't work well at all.